src/math/lp/lp_bound_propagator.h

```diff
@@ -6,19 +6,32 @@
 */
 #pragma once
 #include <utility>
-
 #include "math/lp/lp_settings.h"
 #include "util/uint_set.h"
+#include "math/lp/implied_bound.h"
+#include <vector>
 namespace lp {
 template <typename T>
+struct my_allocator {
+    using value_type = T;
+
+    T* allocate(std::size_t n) {
+        return static_cast<T*>(memory::allocate(n * sizeof(T)));
+    }
+
+    void deallocate(T* p, std::size_t n) {
+        memory::deallocate(p);
+    }
+};    
+template <typename T>
 class lp_bound_propagator {
 	uint_set m_visited_rows;
     // these maps map a column index to the corresponding index in ibounds
-    std::unordered_map<unsigned, unsigned> m_improved_lower_bounds;
-    std::unordered_map<unsigned, unsigned> m_improved_upper_bounds;
+    u_map<unsigned> m_improved_lower_bounds;
+    u_map<unsigned> m_improved_upper_bounds;
 
     T& m_imp;
-    vector<implied_bound> m_ibounds;
+    std::vector<implied_bound, my_allocator<implied_bound>> m_ibounds;
```

Comment:
I will be moving this as a general definition to util/vector in a separate pull request.